Update EIP-7723: update SFI definition based on ACDT#74 (#11475)
ethereum/EIPs: EIPS/eip-7723.md

### Scheduled for Inclusion 

> An EIP should move to SFI when:

Reviewer: How about something along the lines of the following? I feel "intent" is a strong factor in SFI-ing an EIP; we SFI the headliner, long before the spec has adequately been hardened.

```suggestion
An EIP can be moved to SFI when core developers:
1. Agree upon a strong intent to include the EIP in the next network upgrade.
2. Agree that an EIP's specifications and implementations have reached a certain level of maturity.

I.e., an SFI status signals that the EIP is on track for inclusion barring unforeseen issues.

The following critieria can be applied to ascertain whether an EIP is mature (stable) enough to move to SFI:
```

> - It has been included in a devnet that has demonstrated stability (e.g., high participation, no critical bugs for ≥1 week)

Reviewer: ```suggestion
- It has been included in a devnet that has demonstrated stability (e.g., high participation, no critical bugs for ≥1 week).
```

> - The EIP specification is finalized (no placeholder values or pending design decisions)

Reviewer: As-is, this criteria feels slightly too strict. Specs often evolve throughout the process as knowledge is gained, e.g., as other EIPs are subsequently added to future devnets.
```suggestion
- The EIP specification is deemed close to final (no placeholder values or pending design decisions).
```

> - It has minimal interactions with other CFI EIPs, OR those interactions have been tested in the same devnet

Reviewer: ```suggestion
- It has minimal interactions with other CFI EIPs, OR those interactions have been tested in the same devnet.
```

> - Testing coverage is adequate (hive/EEST passing, no known divergence across clients)

Reviewer: I think this can be made more consensus friendly:
```suggestion
- Testing coverage is adequate (clients pass consensus tests, no known divergence across clients).
```

`Scheduled for Inclusion` signals that the EIP is on track for inclusion barring unforeseen issues. EIPs with incomplete specs or untested interactions should remain CFI until these conditions are met. The latest Upgrade Devnet must contain all `Scheduled for Inclusion` Core EIPs.

An EIP **MAY** be moved from `Scheduled for Inclusion` to `Declined for Inclusion` if circumstances necessitate removal, as agreed by client teams. An EIP **MAY** also be moved from `Scheduled for Inclusion` to `Considered for Inclusion` if client teams remain in favor of including the EIP in the network upgrade but cannot commit to including it in the **next** Upgrade Devnet.

> An EIP **MUST** have a Python implementation accompanied by tests in [execution-specs](https://github.com/ethereum/execution-specs/blob/78fb726158c69d8fa164e28f195fabf6ab59b915/README.md), submitted as an open PR or merged to the `devnets/upgradeName/version` branch of the repository. Client developers **MAY** decide to allow an EIP to be moved to `Scheduled for Inclusion` without an [execution-specs](https://github.com/ethereum/execution-specs/blob/78fb726158c69d8fa164e28f195fabf6ab59b915/README.md) implementation, but the tests are strictly mandatory.

Reviewer: The execution-specs [README](https://github.com/ethereum/execution-specs/blob/5cdb055e069e246a877c8aa018079eb0b9093f57/README.md) just got a large revamp, can we bump the ref here?
```suggestion
An EIP **MUST** have a Python implementation accompanied by tests in [execution-specs](https://github.com/ethereum/execution-specs/blob/5cdb055e069e246a877c8aa018079eb0b9093f57/README.md), submitted as an open PR or merged to the `devnets/upgradeName/version` branch of the repository. Client developers **MAY** decide to allow an EIP to be moved to `Scheduled for Inclusion` without an [execution-specs](https://github.com/ethereum/execution-specs/blob/5cdb055e069e246a877c8aa018079eb0b9093f57/README.md) implementation, but the tests are strictly mandatory.
```

I think we're forced by EIP-1 to provide a permalink here? A simple link to the repo would be better, imo, but this is fine.

> Any updates to an EIP that is already at this stage **MUST** be accompanied by appropriate updates to its implementation and tests in [execution-specs](https://github.com/ethereum/execution-specs/blob/78fb726158c69d8fa164e28f195fabf6ab59b915/README.md) if deemed necessary by client developers.

Reviewer: Same README ref bump:

```suggestion
Any updates to an EIP that is already at this stage **MUST** be accompanied by appropriate updates to its implementation and tests in [execution-specs](https://github.com/ethereum/execution-specs/blob/5cdb055e069e246a877c8aa018079eb0b9093f57/README.md) if deemed necessary by client developers.
```
